addons/binding/org.openhab.binding.nikohomecontrol/src/main/java/org/openhab/binding/nikohomecontrol/internal/discovery/ThingDiscoveryService.java

```diff
@@ -0,0 +1,126 @@
+/**
+ * Copyright (c) 2014-2016 by the respective copyright holders.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.nikohomecontrol.internal.discovery;
+
+import static org.openhab.binding.nikohomecontrol.NikoHomeControlBindingConstants.*;
+
+import java.util.Hashtable;
+
+import org.eclipse.smarthome.config.discovery.AbstractDiscoveryService;
+import org.eclipse.smarthome.config.discovery.DiscoveryResult;
+import org.eclipse.smarthome.config.discovery.DiscoveryResultBuilder;
+import org.eclipse.smarthome.config.discovery.DiscoveryService;
+import org.eclipse.smarthome.core.thing.ThingUID;
+import org.openhab.binding.nikohomecontrol.NikoHomeControlBindingConstants;
+import org.openhab.binding.nikohomecontrol.handler.NikoHomeControlBridgeHandler;
+import org.openhab.binding.nikohomecontrol.internal.protocol.NikoHomeControlCommunication;
+import org.osgi.framework.BundleContext;
+import org.osgi.framework.ServiceRegistration;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * If a Niko Home Control bridge is added or if the user scans manually for things this {@link ThingDiscoveryService}
+ * is used to return Niko Home Control Actions as things to the framework.
+ *
+ * @author Mark Herwege
+ */
+public class ThingDiscoveryService extends AbstractDiscoveryService {
```

Comment:
Changed the name of the class and revisited the logic of registering and removing the registration. It turns out I missed removing the registration when the bridge is removed.